pkg/cmd/checkapilb/cmd.go

```diff
@@ -0,0 +1,173 @@
+package checkapilb
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"time"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog/v2"
+
+	configv1client "github.com/openshift/client-go/config/clientset/versioned"
+	"github.com/openshift/library-go/pkg/config/client"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var timeout = 5 * time.Minute
+
+// checkAPILBOpts holds values to drive the check-api-server command.
+type checkAPILBOpts struct {
+	ctx        context.Context
+	KubeConfig string
+
+	RestConfig *rest.Config
+	KubeClient kubernetes.Interface
+}
+
+// NewRenderCommand creates a check-api-server command.
+func NewCheckAPILBCommand(ctx context.Context) *cobra.Command {
+	return newCheckAPILBCommand(ctx)
+}
+
+func newCheckAPILBCommand(ctx context.Context, testOverrides ...func(*checkAPILBOpts)) *cobra.Command {
+	renderOpts := checkAPILBOpts{
+		ctx: ctx,
+	}
+	for _, f := range testOverrides {
+		f(&renderOpts)
+	}
+	cmd := &cobra.Command{
+		Use:   "check-api-lb",
+		Short: "Verify that API loadbalancer can serve API without bootstrap node",
+		Run: func(cmd *cobra.Command, args []string) {
+			if err := renderOpts.Validate(); err != nil {
+				klog.Fatal(err)
+			}
+			if err := renderOpts.Complete(); err != nil {
+				klog.Fatal(err)
+			}
+			if err := renderOpts.Run(); err != nil {
+				klog.Fatal(err)
+			}
+		},
+	}
+
+	renderOpts.AddFlags(cmd.Flags())
+
+	return cmd
+}
+
+func (o *checkAPILBOpts) AddFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&o.KubeConfig, "kubeconfig", o.KubeConfig, "kubeconfig file or empty")
+}
+
+// Validate verifies the inputs.
+func (r *checkAPILBOpts) Validate() error {
+	restConfig, err := client.GetKubeConfigOrInClusterConfig(r.KubeConfig, nil)
+	if err != nil {
+		klog.Fatalf("invalid kubeconfig set: %v", err)
+	}
+	r.RestConfig = rest.CopyConfig(restConfig)
+
+	r.KubeClient = kubernetes.NewForConfigOrDie(restConfig)
+	return nil
+}
+
+// Complete fills in missing values before command execution.
+func (r *checkAPILBOpts) Complete() error {
+	return nil
+}
+
+// Run contains the logic of the check-api-server command.
+func (r *checkAPILBOpts) Run() error {
+	pollCtx, pollCancel := context.WithTimeout(context.Background(), timeout)
+	defer pollCancel()
+
+	attempt := 0
+	return wait.PollImmediateUntil(time.Second, func() (done bool, err error) {
+		retryCtx, retryCancel := context.WithTimeout(context.Background(), time.Second)
```

Comment:
A 1 second context timeout seems too short.

Reply:
Right, I can make it 5 secs just to be safe